src/App.tsx

```diff
@@ -1,58 +1,112 @@
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <Counter />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <span>
-          <span>Learn </span>
-          <a
-            className="App-link"
-            href="https://reactjs.org/"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            React
-          </a>
-          <span>, </span>
-          <a
-            className="App-link"
-            href="https://redux.js.org/"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Redux
-          </a>
-          <span>, </span>
-          <a
-            className="App-link"
-            href="https://redux-toolkit.js.org/"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Redux Toolkit
-          </a>
-          ,<span> and </span>
-          <a
-            className="App-link"
-            href="https://react-redux.js.org/"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            React Redux
-          </a>
-        </span>
-      </header>
-    </div>
-  );
+    const [showModalA, setShowModalA] = useState(window.location.hash === '#ModalA');
+    const [showModalB, setShowModalB] = useState(window.location.hash === '#ModalB');
+    const [onlyEven, setOnlyEven] = useState(false);
+
+    const openModalAHandler = () => {
+        setShowModalA(true);
+        window.location.hash = '#ModalA';
+    }
+
+    const openModalBHandler = () => {
+        setShowModalB(true);
+        window.location.hash = '#ModalB';
+    }
+
+    const closeModalAHandler = () => {
+        setShowModalA(false)
+        window.location.hash = ''
+    }
+
+    const closeModalBHandler = () => {
+        setShowModalB(false)
+        window.location.hash = '';
+    }
+
+    const allContactsClickHandler = () => {
+        if (showModalA) return;
+
+        closeModalBHandler();
+        openModalAHandler();
+    }
+
+    const usContactsClickHandler = () => {
+        if (showModalB) return;
+
+        closeModalAHandler();
+        openModalBHandler();
+    }
+
+    return (
+        <div className={styles.app}>
+            <div className={styles.btnContainer}>
+                <CustomButton
+                    onClick={openModalAHandler}
+                    variant={'purple'}
+                    style={{
+                        marginRight: 10
+                    }}
+                >
+                    Button A
+                </CustomButton>
+                <CustomButton
+                    onClick={openModalBHandler}
+                    variant={'orange'}
+                >
+                    Button B
+                </CustomButton>
+            </div>
+            {showModalA &&
+				<Modal
```

Comment:
Aici ar fi indicat sa creezi un modal cu nume specific in care sa parametrizezi doar ce e diferit. Asa eviti cod duplicat si faci codul mai readable. De asemenea ca un best practice ar trebui sa lasi conditia " {showModalA && " pe modalul generic